src/lib/mina_base/zkapp_command.ml

```diff
@@ -1238,17 +1238,15 @@ let zkapp_command_list (t : t) : Account_update.t list =
 let fee_excess (t : t) =
   Fee_excess.of_single (fee_token t, Currency.Fee.Signed.of_unsigned (fee t))
 
+(* returned statuses same as input status, except always `Applied` for fee payer *)
 let accounts_accessed (t : t) (status : Transaction_status.t) =
-  match status with
-  | Applied ->
-      Call_forest.fold t.account_updates
-        ~init:[ fee_payer t ]
-        ~f:(fun acc p -> Account_update.account_id p :: acc)
-      |> List.rev |> List.stable_dedup
-  | Failed _ ->
-      [ fee_payer t ]
-
-let accounts_referenced (t : t) = accounts_accessed t Applied
+  let init = [ (fee_payer t, Transaction_status.Applied) ] in
+  Call_forest.fold t.account_updates ~init ~f:(fun acc p ->
+      (Account_update.account_id p, status) :: acc )
```

Comment:
Can we use polymorphic variants (`Applied` and `Failed`) instead of transaction status? The transaction status for the failure case has failures associated with all the updates. It can be misleading when associated with one update. Since we don't use the specific failures passed here, might be good to omit those.

Or we have to split the failures and attach the failure associated with each account update (either a failure or `Cancelled` status). I'm in favor of just passing `Applied` or `Failed` since we don't need anymore details for now

Reply:
Made it `Accessed | Not_accessed`